### ISSUE TITLE
'ph_with_text' not inheriting text styling/format

### ISSUE THREAD

**Issue by eknud** [davidgohel/officer#221]
Beginning some point between last week and this week, when generating a PowerPoint for Mac file (pptx), 'ph_with_text' began producing text that does not conform to the text styling/format of the placeholder (whereas prior to this issue, the text would adopt whatever styling was present in the placeholder text). No updates appear to have been made to PowerPoint.

```
sessioninfo::session_info()
─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       macOS Mojave 10.14.5        
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2019-07-15                  

─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package      * version  date       lib source        
 ash            1.0-15   2015-09-01 [1] CRAN (R 3.5.0)
 assertthat     0.2.1    2019-03-21 [1] CRAN (R 3.5.2)
 backports      1.1.4    2019-04-10 [1] CRAN (R 3.5.2)
 base64enc      0.1-3    2015-07-28 [1] CRAN (R 3.5.0)
 bayestestR     0.2.2    2019-06-20 [1] CRAN (R 3.5.2)
 bmp            0.3      2017-09-11 [1] CRAN (R 3.5.0)
 boot           1.3-22   2019-04-02 [1] CRAN (R 3.5.2)
 broom          0.5.2    2019-04-07 [1] CRAN (R 3.5.2)
 cellranger     1.1.0    2016-07-27 [1] CRAN (R 3.5.0)
 cli            1.1.0    2019-03-19 [1] CRAN (R 3.5.2)
 cmprsk         2.2-7    2014-06-17 [1] CRAN (R 3.5.0)
 codetools      0.2-16   2018-12-24 [1] CRAN (R 3.5.2)
 colorspace     1.4-1    2019-03-18 [1] CRAN (R 3.5.2)
 cowplot        0.9.4    2019-01-08 [1] CRAN (R 3.5.2)
 crayon         1.3.4    2017-09-16 [1] CRAN (R 3.5.0)
 data.table   * 1.12.2   2019-04-07 [1] CRAN (R 3.5.2)
 DBI          * 1.0.0    2018-05-02 [1] CRAN (R 3.5.0)
 digest         0.6.19   2019-05-20 [1] CRAN (R 3.5.2)
 dplyr        * 0.8.2    2019-06-29 [1] CRAN (R 3.5.2)
 emmeans        1.3.5.1  2019-06-26 [1] CRAN (R 3.5.2)
 estimability   1.3      2018-02-11 [1] CRAN (R 3.5.0)
 extrafont      0.17     2014-12-08 [1] CRAN (R 3.5.0)
 extrafontdb    1.0      2012-06-11 [1] CRAN (R 3.5.0)
 forcats      * 0.4.0    2019-02-17 [1] CRAN (R 3.5.2)
 foreign        0.8-71   2018-07-20 [1] CRAN (R 3.5.2)
 generics       0.0.2    2018-11-29 [1] CRAN (R 3.5.0)
 ggalt        * 0.4.0    2017-02-15 [1] CRAN (R 3.5.0)
 ggeffects      0.11.0   2019-07-01 [1] CRAN (R 3.5.2)
 ggplot2      * 3.2.0    2019-06-16 [1] CRAN (R 3.5.2)
 ggpubr       * 0.2.1    2019-06-23 [1] CRAN (R 3.5.2)
 ggrepel      * 0.8.1    2019-05-07 [1] CRAN (R 3.5.2)
 ggsignif       0.5.0    2019-02-20 [1] CRAN (R 3.5.2)
 glmmTMB        0.2.3    2019-01-11 [1] CRAN (R 3.5.2)
 glue           1.3.1    2019-03-12 [1] CRAN (R 3.5.2)
 googlesheets * 0.3.0    2018-06-29 [1] CRAN (R 3.5.0)
 gridExtra    * 2.3      2017-09-09 [1] CRAN (R 3.5.0)
 gtable         0.3.0    2019-03-25 [1] CRAN (R 3.5.2)
 haven          2.1.0    2019-02-19 [1] CRAN (R 3.5.2)
 hms            0.4.2    2018-03-10 [1] CRAN (R 3.5.0)
 htmltools      0.3.6    2017-04-28 [1] CRAN (R 3.5.0)
 httr         * 1.4.0    2018-12-11 [1] CRAN (R 3.5.0)
 igraph         1.2.4.1  2019-04-22 [1] CRAN (R 3.5.2)
 imager         0.41.2   2019-01-23 [1] CRAN (R 3.5.2)
 insight        0.4.0    2019-07-01 [1] CRAN (R 3.5.2)
 jpeg           0.1-8    2014-01-23 [1] CRAN (R 3.5.0)
 jsonlite     * 1.6      2018-12-07 [1] CRAN (R 3.5.0)
 KernSmooth     2.23-15  2015-06-29 [1] CRAN (R 3.5.2)
 km.ci          0.5-2    2009-08-30 [1] CRAN (R 3.5.0)
 KMsurv         0.1-5    2012-12-03 [1] CRAN (R 3.5.0)
 knitr          1.23     2019-05-18 [1] CRAN (R 3.5.2)
 labeling       0.3      2014-08-23 [1] CRAN (R 3.5.0)
 lattice        0.20-38  2018-11-04 [1] CRAN (R 3.5.2)
 lazyeval       0.2.2    2019-03-15 [1] CRAN (R 3.5.2)
 lme4           1.1-21   2019-03-05 [1] CRAN (R 3.5.2)
 lubridate    * 1.7.4    2018-04-11 [1] CRAN (R 3.5.0)
 magrittr     * 1.5      2014-11-22 [1] CRAN (R 3.5.0)
 maps           3.3.0    2018-04-03 [1] CRAN (R 3.5.0)
 MASS           7.3-51.4 2019-03-31 [1] CRAN (R 3.5.2)
 Matrix         1.2-17   2019-03-22 [1] CRAN (R 3.5.2)
 minqa          1.2.4    2014-10-09 [1] CRAN (R 3.5.0)
 mnormt         1.5-5    2016-10-15 [1] CRAN (R 3.5.0)
 modelr         0.1.4    2019-02-18 [1] CRAN (R 3.5.2)
 multcomp       1.4-10   2019-03-05 [1] CRAN (R 3.5.2)
 munsell        0.5.0    2018-06-12 [1] CRAN (R 3.5.0)
 mvtnorm        1.0-11   2019-06-19 [1] CRAN (R 3.5.2)
 namely       * 0.0.1    2019-03-14 [1] local         
 nlme           3.1-140  2019-05-12 [1] CRAN (R 3.5.2)
 nloptr         1.2.1    2018-10-03 [1] CRAN (R 3.5.0)
 officer      * 0.3.5    2019-06-23 [1] CRAN (R 3.5.2)
 openxlsx     * 4.1.0.1  2019-05-28 [1] CRAN (R 3.5.2)
 performance    0.2.0    2019-06-04 [1] CRAN (R 3.5.2)
 pillar         1.4.2    2019-06-29 [1] CRAN (R 3.5.2)
 pkgconfig      2.0.2    2018-08-16 [1] CRAN (R 3.5.0)
 plyr         * 1.8.4    2016-06-08 [1] CRAN (R 3.5.0)
 png            0.1-7    2013-12-03 [1] CRAN (R 3.5.0)
 proj4          1.0-8    2012-08-05 [1] CRAN (R 3.5.0)
 psych          1.8.12   2019-01-12 [1] CRAN (R 3.5.2)
 purrr        * 0.3.2    2019-03-15 [1] CRAN (R 3.5.2)
 R.methodsS3  * 1.7.1    2016-02-16 [1] CRAN (R 3.5.0)
 R.oo         * 1.22.0   2018-04-22 [1] CRAN (R 3.5.0)
 R6             2.4.0    2019-02-14 [1] CRAN (R 3.5.2)
 RColorBrewer   1.1-2    2014-12-07 [1] CRAN (R 3.5.0)
 Rcpp           1.0.1    2019-03-17 [1] CRAN (R 3.5.2)
 readbitmap     0.1.5    2018-06-27 [1] CRAN (R 3.5.0)
 readr        * 1.3.1    2018-12-21 [1] CRAN (R 3.5.0)
 readxl         1.3.1    2019-03-13 [1] CRAN (R 3.5.2)
 reshape2       1.4.3    2017-12-11 [1] CRAN (R 3.5.0)
 rlang          0.4.0    2019-06-25 [1] CRAN (R 3.5.2)
 RPostgreSQL  * 0.6-2    2017-06-24 [1] CRAN (R 3.5.0)
 rstudioapi     0.10     2019-03-19 [1] CRAN (R 3.5.2)
 Rttf2pt1       1.3.7    2018-06-29 [1] CRAN (R 3.5.0)
 rvest          0.3.4    2019-05-15 [1] CRAN (R 3.5.2)
 sandwich       2.5-1    2019-04-06 [1] CRAN (R 3.5.2)
 scales       * 1.0.0    2018-08-09 [1] CRAN (R 3.5.0)
 sessioninfo    1.1.1    2018-11-05 [1] CRAN (R 3.5.0)
 sjlabelled     1.1.0    2019-06-06 [1] CRAN (R 3.5.2)
 sjmisc         2.8.1    2019-06-17 [1] CRAN (R 3.5.2)
 sjPlot       * 2.6.3    2019-04-27 [1] CRAN (R 3.5.2)
 sjstats        0.17.5   2019-06-04 [1] CRAN (R 3.5.2)
 stringi        1.4.3    2019-03-12 [1] CRAN (R 3.5.2)
 stringr      * 1.4.0    2019-02-10 [1] CRAN (R 3.5.2)
 survival     * 2.44-1.1 2019-04-01 [1] CRAN (R 3.5.2)
 survminer    * 0.4.4    2019-05-21 [1] CRAN (R 3.5.2)
 survMisc       0.5.5    2018-07-05 [1] CRAN (R 3.5.0)
 TH.data        1.0-10   2019-01-21 [1] CRAN (R 3.5.2)
 tibble       * 2.1.3    2019-06-06 [1] CRAN (R 3.5.2)
 tidyr        * 0.8.3    2019-03-01 [1] CRAN (R 3.5.2)
 tidyselect     0.2.5    2018-10-11 [1] CRAN (R 3.5.0)
 tidyverse    * 1.2.1    2017-11-14 [1] CRAN (R 3.5.0)
 tiff           0.1-5    2013-09-04 [1] CRAN (R 3.5.0)
 TMB            1.7.15   2018-11-09 [1] CRAN (R 3.5.0)
 uuid           0.1-2    2015-07-28 [1] CRAN (R 3.5.0)
 withr          2.1.2    2018-03-15 [1] CRAN (R 3.5.0)
 xfun           0.8      2019-06-25 [1] CRAN (R 3.5.2)
 xml2           1.2.0    2018-01-24 [1] CRAN (R 3.5.0)
 xtable         1.8-4    2019-04-21 [1] CRAN (R 3.5.2)
 yaml           2.2.0    2018-07-25 [1] CRAN (R 3.5.0)
 zip            2.0.2    2019-05-13 [1] CRAN (R 3.5.2)
 zoo            1.8-6    2019-05-28 [1] CRAN (R 3.5.2)

[1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
```

**Comment by davidgohel** [comment]
Can you use `ph_with` instead. `ph_with_text` will be deprecated later this year and `ph_with` should inherits text style as expected

**Comment by eknud** [comment]
@davidgohel Thank you! This appears to have worked! Any explanation for what caused this?

**Comment by john-harrold** [comment]
In case anyone runs into this here is an example that worked for me. To get the equivalent of this:
` ph_with_text(x=tmprpt, type='body', index=3, str="My Title")`

I replaced it with this: 
`ph_with(x=tmprpt, location = ph_location_type(type = "body"),   index=3, value="My Title")`

Basically I had to start passing the type information using ph_location_type and I also had to change the str input to value. 

I was also running into a similar problem building lists on the fly using `ph_empty` I also had to pass the type information through using ph_location_type. For example I would change this:
`ph_empty(x=tmprpt, type="body", index=3)`

To something like this:
`ph_empty(x=tmprpt, location=ph_location_type(type="body"), index=3)`

Then I could use `ph_add_par` and `ph_add_text` to build a list.

**Comment by davidgohel** [comment]
@eknud This comes from users that did want something easier to use when adding content in slides. That's why I had to create `ph_with` that works with `location` argument that is expecting a `ph_location*` specification. 

@john-harrold If possible, I'd like people stop using `ph_empty`, `ph_add_par` and `ph_add_text` to add new elements. There are other functions easier to use to add content in new shape. See https://davidgohel.github.io/officer/articles/powerpoint.html#add-content-in-a-placeholder and also https://davidgohel.github.io/officer/articles/powerpoint.html#supported-contents

`ph_add_par` and `ph_add_text` are to be used to add something to an **existing** content, i.e. something with existing text that needs to be preserved. I added `ph_empty` to help a user long time ago and I was not expecting people to use this as a replacement of `ph_with` functions ;)

I will update and complete the doc soon.
